From 9bac999ae1b26c0c61038a0817856b93014798b6 Mon Sep 17 00:00:00 2001
Message-Id: <9bac999ae1b26c0c61038a0817856b93014798b6.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Tue, 13 Jan 2015 15:50:11 +0530
Subject: [PATCH 43/95] MUST_REBASE [VPG]: video/adf/intel: Update the DSI
 disable path to support dual link panel disabling

We need to program both port registers during dual link disable path.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

For: GMINL-5276
Change-Id: Ic798887ca88eedd340740e78f6b492dc12d5ac3a
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_dsi_port.c    |    2 +-
 drivers/video/adf/intel/core/vlv/vlv_pipeline.c    |   20 +++++++++++++-------
 .../adf/intel/include/core/vlv/vlv_dsi_port.h      |    2 +-
 3 files changed, 15 insertions(+), 9 deletions(-)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_dsi_port.c b/drivers/video/adf/intel/core/vlv/vlv_dsi_port.c
index 491e0344..f12cc48 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_dsi_port.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_dsi_port.c
@@ -28,7 +28,7 @@ u32 vlv_dsi_port_is_cmd_mode(struct vlv_dsi_port *port)
 	return port->operation_mode == DSI_DBI;
 }
 
-u32  vlv_dsi_port_wait_for_fifo_empty(struct vlv_dsi_port *port)
+u32 vlv_dsi_port_wait_for_fifo_empty(struct vlv_dsi_port *port)
 {
 	u32 mask;
 
diff --git a/drivers/video/adf/intel/core/vlv/vlv_pipeline.c b/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
index 22a6997..af12494 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
@@ -314,15 +314,21 @@ u32 vlv_dsi_post_pipeline_off(struct intel_pipeline *pipeline)
 {
 	struct vlv_pipeline *disp = to_vlv_pipeline(pipeline);
 	struct vlv_pll *pll = &disp->pll;
-	struct vlv_dsi_port *dsi_port = &disp->port.dsi_port[pll->port_id];
-	u32 err = 0;
-	err = vlv_dsi_port_wait_for_fifo_empty(dsi_port);
-
-	err = vlv_dsi_port_clear_device_ready(dsi_port);
+	struct vlv_dsi_port *dsi_port = NULL;
+	struct dsi_pipe *dsi_pipe = NULL;
+	struct dsi_context *dsi_ctx = NULL;
+	enum port port;
 
-	err = vlv_dsi_pll_disable(pll);
+	dsi_pipe = &disp->gen.dsi;
+	dsi_ctx = &dsi_pipe->config.ctx;
+	for_each_dsi_port(port, dsi_ctx->ports) {
+		dsi_port = &disp->port.dsi_port[port];
+		vlv_dsi_port_wait_for_fifo_empty(dsi_port);
+		vlv_dsi_port_clear_device_ready(dsi_port);
+	}
+	vlv_dsi_pll_disable(pll);
 
-	return err;
+	return 0;
 }
 
 /* generic function to be called for any operations after disable is done */
diff --git a/drivers/video/adf/intel/include/core/vlv/vlv_dsi_port.h b/drivers/video/adf/intel/include/core/vlv/vlv_dsi_port.h
index 56dea3e..706ac0e 100644
--- a/drivers/video/adf/intel/include/core/vlv/vlv_dsi_port.h
+++ b/drivers/video/adf/intel/include/core/vlv/vlv_dsi_port.h
@@ -70,7 +70,7 @@ u32 vlv_dsi_port_disable(struct vlv_dsi_port *port,
 		struct dsi_config *config);
 bool vlv_dsi_port_set_device_ready(struct vlv_dsi_port *port);
 bool vlv_dsi_port_clear_device_ready(struct vlv_dsi_port *port);
-u32  vlv_dsi_port_wait_for_fifo_empty(struct vlv_dsi_port *port);
+u32 vlv_dsi_port_wait_for_fifo_empty(struct vlv_dsi_port *port);
 bool vlv_dsi_port_can_be_disabled(struct vlv_dsi_port *port);
 u32 vlv_dsi_port_is_vid_mode(struct vlv_dsi_port *port);
 u32 vlv_dsi_port_is_cmd_mode(struct vlv_dsi_port *port);
-- 
1.7.9.5

